FrostCord/FrostCord
# Invite Links

# Invite Links
This PR adds invite links via a splash page and an in-message embed, both feature buttons to join the server, which when clicked, will navigate you to the respective server and channel the invite goes go.

# API Changes:
- The REST api for creating invites has moved to `/api/v1/[serverId]/[channelId]/invite/create`
- Invite creation now **requires** a channel id
- Index now takes in a `c` query param for navigation directly to the channel. This does NOT allow access to channels the user has no permissions to see (or are from a server they are not a member of)
- Servers are now sorted in alphabetical order
- Marquee has been extracted to its own component, `OverflowMarquee`

# Fixes:
- Fixes a bug where inserts to server users were received across all clients and updated every connected client's serverlist with the new entry
- Fixes some weird Marquee issues where the size of the span of marquee text and the static text were different sizes causing render bugs

# Testing
- Verify invites don't let you use their buttons for servers you're already a member of
- Verify navigating to a channel (using `/?c=<channel_id>`) redirects to the default page if the id is invalid or from a channel you do not have access to
- Verify clicking the join button on an invite (either the splash page or embed) will navigate to the target channel once joined
- Verify servers are sorted in alpha order regardless of case
